src/algebra/order/field_pow.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.group_with_zero.power
+import algebra.order.field
+/-!
+# Powers of elements of linear ordered fields
+
+Some results on integer powers of elements of a linear ordered field.
+These results are in their own file because they depend both on
+`linear_ordered_field` and on `zpow`; neither should be obviously
+an ancestor of the other.
```

Comment:
Moving this to `algebra.order.field` would reintroduce a cycle (since `data.nat.cast` imports `algebra.order.field`). I want to break that dependency too, so I could remake this PR to break both dependencies at the same time.

Reply:
I guess what you want is not referring to `zpow`, but rather to its API.

Reply:
> I could remake this PR to break both dependencies at the same time.

The PR that breaks the other dependency is #14890, by the way.

Reply:
Maybe then
```suggestion
`linear_ordered_field` and on the API for `zpow`; neither should be obviously
an ancestor of the other.
```

Reply:
> I want to break that last dependency too, so I could remake this PR to break both dependencies at the same time.

That's now #14890, right? Does that mean this now _does_ fit in algebra.order.field?

Reply:
Indeed, we could move these lemmas to `algebra.order.field` after applying the changes in (this PR and) #14890. Should I do so in #14890?

Reply:
Sure, that seems reasonable to me.